flocker/acceptance/testtools.py

```diff
@@ -703,6 +729,59 @@ def current_nodes(self):
         return request
 
     @log_method
+    def install_flocker_version(self, package_source):
+        """
+        Change the version of flocker installed on all of the nodes to the
+        version indicated by `package_source`.
+        """
+        control_node_address = self.control_node.public_address
+        all_cluster_nodes = set([x.public_address for x in self.nodes] +
+                                [control_node_address])
+        distribution = self.distribution
+
+        def get_flocker_version():
+            # Retry getting the flocker version for 10 seconds. Flocker might
+            # not be running yet on the control node when this is called.
+            d = timeout(
+                self.reactor,
+                retry_failure(
+                    self.reactor,
+                    self.client.version
+                ),
+                10
+            )
+            d.addCallback(lambda v: str(v.get('flocker')) or None)
+            return d
+
+        d = get_flocker_version()
+
+        # If we fail to get the current version, assume we must reinstall
+        # flocker.
+        d.addErrback(write_failure)
+
+        def reinstall_if_needed(v):
+            if v and v == package_source.version:
+                return v
+            return reinstall_flocker_at_version(
+                reactor, all_cluster_nodes, control_node_address,
+                package_source, distribution)
```

Comment:
:question: I looked at the name of this function and expected it to require a ``version`` argument.